stdlib/_operator.pyi

```diff
@@ -77,11 +78,9 @@ def delitem(__a: MutableSequence[Any], __b: slice) -> None: ...
 @overload
 def delitem(__a: MutableMapping[_K, Any], __b: _K) -> None: ...
 @overload
-def getitem(__a: Sequence[_T], __b: SupportsIndex) -> _T: ...
+def getitem(__a: SupportsGetItem[_K, _V], __b: _K) -> _V: ...
```

Comment:
Won't this largely mask the `slice` overload in practice? E.g. if we call `getitem([1, 2, 3], slice(1))`, we should get a Sequence back, but I think this overload will be matched and we'll get a slice instead.

Reply:
I was also worried about this, but I tried it out and it doesn't seem to be the case: https://mypy-play.net/?mypy=latest&python=3.10&gist=184cb82e0026c4bd0e9a6d5fa47eb1a1

Might be worth adding a test case, though

Reply:
I think a slice will never match the first overload (at least with current typing definitions) because:
```
Argument of type "slice" cannot be assigned to parameter "__b" of type "_K@getitem" in function "getitem"
  Type "slice" cannot be assigned to type "SupportsIndex"
    "slice" is incompatible with protocol "SupportsIndex"
      "__index__" is not present
```

Still I think it makes sense to put the more precise overload first. Out of good practice.